java-client/src/main/java/org/opensearch/client/opensearch/ingest/PutPipelineRequest.java

```diff
@@ -481,6 +481,7 @@ protected static void setupPutPipelineRequestDeserializer(ObjectDeserializer<Put
 
         op.add(Builder::meta, JsonpDeserializer.stringMapDeserializer(JsonData._DESERIALIZER), "_meta");
         op.add(Builder::description, JsonpDeserializer.stringDeserializer(), "description");
+        op.add(Builder::id, JsonpDeserializer.stringDeserializer(), "id");
```

Comment:
@patschl the `PutPipelineRequest` does not include `id` as the payload - it is part of the URL and passed as the URL path component. Under which circumstances it is returned in the request payload? Could you please share the example? (sadly the linked issue has no payload sample).

Reply:
Thanks @reta!
I'm not quite sure I understand the purpose of this deserializer correctly if it should strictly deal with the payload of a request.
`id` is a required field for the `PutPipelineRequest`, so deserializing this object will currently fail every time, since the deserializer of course won't set it. Wouldn't this make the deserializer unnecessary if it won't work anyway?

We had a similar issue with the `name` property in the `PutIndexTemplateRequest`, but managed to work around it.

We persist such index template and pipeline definitions as json in our VCS and create the requests on demand, this is the reason for this PR.

Reply:
Thanks @patschl 

> I'm not quite sure I understand the purpose of this deserializer correctly if it should strictly deal with the payload of a request.

The deserializer mirrors serializer: in this particular case, the `id` is not part of the serialized JSON (payload) and as such, should not be deserialized from the payload (since it should be there in the first place).

Reply:
@reta Okay I see, I will close this PR then. Thank you for clearing that up.
There is no way to create this object from a json using this deserializer, but I guess this was never the use case anyway?

Reply:
> There is no way to create this object from a json using this deserializer, but I guess this was never the use case anyway?

Thank you @patschl, I think the question is under which circumstances it should happen, this is a request which naturally only uses one way serialization (when send to the service). But in general - yes, it seems like some models may need more context (URI components) to deserialize completely.